From 74a1c872a49e44d9b1cc5ecde635dabd373b305b Mon Sep 17 00:00:00 2001
From: Tyler Gunn <tgunn@google.com>
Date: Thu, 6 Jan 2022 14:50:22 -0800
Subject: [PATCH] Fix pending intent issue in SipAccountRegistry.

Add immutable flag.

Test: Manual test; build code.
Fixes: 208817618
Change-Id: Iec8e546db809c001bb377406e3c33d5d714ddbed
(cherry picked from commit 7c9b65a4de4540a50a16781e9f55857544453bc2)
Merged-In:Iec8e546db809c001bb377406e3c33d5d714ddbed
---
 .../com/android/services/telephony/sip/SipAccountRegistry.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sip/src/com/android/services/telephony/sip/SipAccountRegistry.java b/sip/src/com/android/services/telephony/sip/SipAccountRegistry.java
index 2845dacfc..6b34f00ed 100644
--- a/sip/src/com/android/services/telephony/sip/SipAccountRegistry.java
+++ b/sip/src/com/android/services/telephony/sip/SipAccountRegistry.java
@@ -236,7 +236,8 @@ public final class SipAccountRegistry {
 
         Intent intent = new Intent(TelecomManager.ACTION_CHANGE_PHONE_ACCOUNTS);
         intent.setFlags(Intent.FLAG_RECEIVER_INCLUDE_BACKGROUND);
-        PendingIntent pendingIntent = PendingIntent.getActivity(context, 0, intent, 0);
+        PendingIntent pendingIntent = PendingIntent.getActivity(context, 0, intent,
+                PendingIntent.FLAG_IMMUTABLE);
 
         Notification.Action action = new Notification.Action.Builder(R.drawable.ic_sim_card,
                 context.getString(R.string.sip_accounts_removed_notification_action),
-- 
2.33.1

